src/System.Management.Automation/FormatAndOutput/common/TableWriter.cs

```diff
@@ -289,22 +289,64 @@ private string[] GenerateTableRow(string[] values, int[] alignment, DisplayCells
                     screenRows = scArray[k].Count;
             }
 
+            // column headers can span multiple rows if the width of the column is shorter than the header text like:
+            //
+            // Long Header2 Head
+            // Head         er3
+            // er
+            // ---- ------- ----
+            // 1    2       3
+            //
+            // To ensure we don't add whitespace to the end, we need to determine the last column in each row with content
+
+            int[] lastColWithContent = new int[screenRows];
```

Comment:
We could use new C# Span:
```c#
Span<int> lastColWithContent = stackalloc int[screenRows]; 
```

Reply:
good suggestion, will change

Reply:
[System.Memory](https://www.nuget.org/packages/System.Memory/4.5.0-preview1-26216-02) is still preview, though

Reply:
I see it is compatible with .NETCoreApp 2.0.

Reply:
It's compatible, but do we want to take dependency on something that's Preview?

Reply:
Seems we have plans to move .Net Core 2.1.0 Preview in days/weeks. If so we can start here and now.

Reply:
If we expect System.Memory to GA when .Net Core 2.1 GA's I think this is fine.  We can revert this line later if this turns out to not be true.

Reply:
https://blogs.msdn.microsoft.com/dotnet/2018/02/02/net-core-2-1-roadmap/

Span<T> is in .Net Core 2.1 roadmap.